src/app.js

```diff
@@ -0,0 +1,27 @@
+var bodyParser = require('body-parser');
+var express = require('express');
+
+var maryStatus = 418;
+var maryResponse = {im: 'a teapot'};
+
+var app = express();
+
+app.use(bodyParser.json());
+
+app.get('/alligators/Mary', function (request, response) {
+    response.set('Content-Type', 'application/json; charset=utf-8');
+    response.status(maryStatus).send(maryResponse);
+});
+
+app.post('/dodgy-backdoor/control-what-mary-does', function (request, response) {
```

Comment:
It is. You can do anything you like in the set up method. For example a database could be updated, a file could be changed, maybe a series of api calls to the server could be made. I was assuming that how this particular example worked was not important, more the mechanism of Ruby triggering the state changes.